workspaceitem-data-upload.md

```diff
@@ -0,0 +1,205 @@
+# WorkspaceItem data of upload sectionType
+[Back to the definition of the workspaceitems endpoint](workspaceitems.md)
+
+The section data represent the data about the user uploaded files
+
+```json
+{
+	"files": [ 
+  	 	{
+  	 		metadata: {
+  	 			"dc.title" : [{value: "sample_file.pdf"}],
+  	 			"dc.description" : [{value: "Description of the sample file"}]
+  	 		},
+  	 		"sizeBytes": 8528,
+			"checkSum": {
+			    "checkSumAlgorithm": "MD5",
+			    "value": "9d8f0f9e369cf12159d47c146c499cf4"
+			},
+  	 		"url": "http://dspace7.4science.it/api/core/bitstreams/00001abf-b2e0-477a-99de-104db7cb6469/content",
```

Comment:
Is there a reason why you did not embed or linked to the corresponding bitstream here? The bitstream REST model already contains all this information.

Reply:
Yes, we cannot embed the bitstream here because the `sections/uploads` is not an HAL resource but a simple (deeply nested) attribute of the workspaceitem.
We have explored the option to make any subsection an HAL resource itself, that at the end would have allowed us to expose such links but the resulting model were to complex.
For instance we will have end with the needs to support the following endpoints

/workspaceitems/:id/sections
/workspaceitems/:id/sections/uploads
/workspaceitems/:id/sections/uploads/1
/workspaceitems/:id/sections/uploads/1/accessConditions
etc.

and we have agreed at the start of the REST project to avoid deeply nested url as much as possible.
In addition, this structure would 
- have made the creation of a patch request to modify multiple aspects of a workspaceitem more tricky requiring most probably some hack of the spec to allow all the operation to be performed against a single workspaceitems endpoint instead than the specific one
- result in a much bigger json file with lot of embedded resource "wrapper"